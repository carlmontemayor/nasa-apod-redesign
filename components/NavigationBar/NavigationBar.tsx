import { ReactElement } from 'react'

// This is a demo of husky, a git tool.
export const NavigationBar = (): ReactElement => {
  return (
    <nav>
      <div className="flex relative">
        <div className="font-bold text-5xl p-4 pr-0">APOD</div>
        <div className="font-bold text-sm center pr-4 pt-10 ">
          (astronomy picture of the day)
        </div>
      </div>
      <div className="font-bold text-sm  bottom-0 right-0 absolute p-2 ">
        Astronomy Picture of the Day
      </div>
    </nav>
  )
}
