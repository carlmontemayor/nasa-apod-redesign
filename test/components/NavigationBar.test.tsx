import * as React from 'react'
import { NavigationBar } from '../../components/NavigationBar/NavigationBar'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<NavigationBar />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the aproppriate title, subtitle, and footer', () => {
    render(<NavigationBar />)
    expect(screen.getByText('APOD')).toHaveTextContent('APOD')
    expect(
      screen.getByText('(astronomy picture of the day)')
    ).toHaveTextContent('(astronomy picture of the day)')
    expect(screen.getByText('Astronomy Picture of the Day')).toHaveTextContent(
      'Astronomy Picture of the Day'
    )
  })
})
