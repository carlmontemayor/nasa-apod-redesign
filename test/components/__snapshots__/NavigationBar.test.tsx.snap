// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <nav>
    <div
      class="flex relative"
    >
      <div
        class="font-bold text-5xl p-4 pr-0"
      >
        APOD
      </div>
      <div
        class="font-bold text-sm center pr-4 pt-10 "
      >
        (astronomy picture of the day)
      </div>
    </div>
    <div
      class="font-bold text-sm  bottom-0 right-0 absolute p-2 "
    >
      Astronomy Picture of the Day
    </div>
  </nav>
</DocumentFragment>
`;
